src/test/java/com/develhope/spring/deals/services/OrderServiceTest.java

```diff
@@ -43,48 +47,114 @@ public void setup() {
 
-
     private static final OrderCreatorDTO DEFAULT_ORDER_CREATOR_DTO = new OrderCreatorDTO(
             true,
-            1,
-            1,
+            1L,
+            1L,
             OrderStatus.PAID,
             true
     );
 
-    private static final Vehicle DEFAULT_VEHICLE = new Vehicle(1);
+    private static final Vehicle DEFAULT_VEHICLE = new Vehicle(1L);
+    private static final User DEFAULT_USER = new User(1L);
+
     private static final VehicleOrderReturnerDTO DEFAULT_VEHICLE_ORDER_RETURNER_DTO = new VehicleOrderReturnerDTO(
-            1,
+            1L,
             VehicleType.CAR,
             "Fiat",
             "Fiorino",
             Colors.WHITE,
-            BigDecimal.valueOf(1000).setScale(1, 1),
+            BigDecimal.valueOf(1000).setScale(1, RoundingMode.HALF_UP),
             UsedFlag.NEW,
             "Motore"
     );
-    private static final User DEFAULT_USER = new User(1);
 
+    private static final Order DEFAULT_ORDER = new Order(
+            1L, 1L, true, OrderStatus.PAID, true, DEFAULT_VEHICLE, DEFAULT_USER
+    );
+
+    private static final OrderResponseDTO DEFAULT_ORDER_RESPONSE_DTO = new OrderResponseDTO(
+            true,
+            DEFAULT_VEHICLE_ORDER_RETURNER_DTO,
+            1L,
+            OrderStatus.PAID,
+            true
+    );
 
     @Test
     void createOrder_successfulTest() {
-        OrderResponseDTO expected = new OrderResponseDTO(
-                true,
-                DEFAULT_VEHICLE_ORDER_RETURNER_DTO,
-                1,
-                OrderStatus.PAID,
-                true
-        );
-
         when(vehicleRepository.findById(DEFAULT_ORDER_CREATOR_DTO.getVehicleId()))
                 .thenReturn(Optional.of(DEFAULT_VEHICLE));
         when(userRepository.findById(DEFAULT_ORDER_CREATOR_DTO.getUserId()))
                 .thenReturn(Optional.of(DEFAULT_USER));
+        when(orderMapper.toEntity(DEFAULT_ORDER_CREATOR_DTO))
+                .thenReturn(DEFAULT_ORDER);
+        when(orderRepository.save(DEFAULT_ORDER))
+                .thenReturn(DEFAULT_ORDER);
+        when(orderMapper.toResponseDTO(DEFAULT_ORDER))
+                .thenReturn(DEFAULT_ORDER_RESPONSE_DTO);
 
         OrderResponseDTO result = orderService.create(DEFAULT_ORDER_CREATOR_DTO);
-        assertEquals(expected.getUserId(), result.getUserId());
+
+        assertNotNull(result);
+        assertEquals(DEFAULT_ORDER_RESPONSE_DTO.getUserId(), result.getUserId());
+        assertEquals(DEFAULT_ORDER_RESPONSE_DTO.getVehicle().getId(), result.getVehicle().getId());
+        assertEquals(DEFAULT_ORDER_RESPONSE_DTO.getOrderStatus(), result.getOrderStatus());
+        assertEquals(DEFAULT_ORDER_RESPONSE_DTO.isDownPayment(), result.isDownPayment());
+        assertEquals(DEFAULT_ORDER_RESPONSE_DTO.isPaid(), result.isPaid());
+    }
+
+    @Test
+    void deleteOrder_successfulTest() {
+        when(vehicleRepository.findById(DEFAULT_ORDER_CREATOR_DTO.getVehicleId()))

```

Comment:
questo test è eccessivamente verboso per un delete e rnon comprendo la parte in cui richiami createOrder e anche dove gli dici doNothing quando la repository deve eliminare tramite id, visto che è ciò che dovrebbe fare